src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
 
 describe(__filename, () => {
-  const render = (moreProps: Partial<PublicProps> = {}) => {
+  type RenderParams = { store?: Store } & Partial<PublicProps> &
+    Partial<DefaultProps>;
+
+  const render = ({
+    store = configureStore(),
+    commentId = null,
+    ...moreProps
+  }: RenderParams = {}) => {
     const props = {
-      comment: null,
-      readOnly: true,
+      addonId: 1,
+      commentId,
+      fileName: null,
+      line: null,
+      readOnly: false,
+      versionId: 2,
       ...moreProps,
     };
-    return shallow(<Comment {...props} />);
+    return shallowUntilTarget(<Comment {...props} />, CommentBase, {
+      shallowOptions: { context: { store } },
+    });
   };
 
-  it('renders a form when there is no comment', () => {
-    const root = render({ comment: null });
+  const dispatchComment = ({
+    store = configureStore(),
+    comment = createFakeExternalComment(),
+    fileName = null,
+    line = null,
+    versionId = 1,
+  } = {}) => {
+    store.dispatch(actions.setComment({ fileName, line, versionId, comment }));
+    return { store };
+  };
 
-    expect(root.find(`.${styles.form}`)).toHaveLength(1);
-    expect(root.find(`.${styles.textarea}`)).toHaveLength(1);
+  const createFakeChangeEvent = ({
+    name,
+    value,
+  }: {
+    name: string;
+    value: string;
+  }) => {
+    return createFakeEvent({ target: { name, value } });
+  };
+
+  const createFakeTextareaRef = (currentProps = {}) => {
+    return {
+      ...React.createRef(),
+      current: {
+        ...document.createElement('textarea'),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        focus: jest.fn(),
+        ...currentProps,
+      },
+    };
+  };
+
+  it('requires a comment when readOnly=true', () => {
+    expect(() => render({ readOnly: true, commentId: null })).toThrow(
+      /initialComment for commentId=null cannot be empty/,
+    );
+  });
+
+  it('throws when a commentId is not mapped to a comment', () => {
+    expect(() => render({ readOnly: false, commentId: 1 })).toThrow(
+      /No comment was mapped/,
+    );
   });
 
-  it('renders a comment when provided', () => {
-    const comment = createFakeComment();
+  it('renders a comment when readOnly=true', () => {
+    const content = 'Example of a comment';
+    const comment = createFakeExternalComment({ comment: content });
+    const { store } = dispatchComment({ comment });
 
-    const root = render({ comment });
+    const root = render({ commentId: comment.id, store, readOnly: true });
 
     expect(root.find(`.${styles.form}`)).toHaveLength(0);
     expect(root.find(`.${styles.comment}`)).toHaveLength(1);
-    expect(root.find(`.${styles.comment}`).html()).toContain(comment.content);
+    expect(root.find(`.${styles.comment}`).html()).toContain(content);
     expect(root.find(FontAwesomeIcon)).toHaveLength(1);
   });
 
   it('sanitizes the content of a comment', () => {
```

Comment:
Nit: This only applies to `readOnly: true`. I wonder if it would be worthwhile to add that to the test description, or perhaps add a `describe` block for the `readOnly` tests.

Reply:
I don't think it's important to add `readOnly: true` to the spec because it's not a parameter that the feature depends on. To put it another way: the sanitizer feature doesn't depend on the `readOnly` prop for anything.